lib/filter-stream.js

```diff
@@ -168,39 +223,52 @@ function FilterStream( blockMap, options ) {
    */
   _transform( chunk, _, next ) {
 
-    this._buffers.push( chunk )
-    this._bytes += chunk.length
+    debug( 'position', this.position )
+    debug( 'chunk', chunk.length, chunk )
 
-    if( this._bytes < this.blockSize )
-      return next()
+    this.bytesRead += chunk.length
+    this.blocksRead += chunk.length / this.blockSize
 
-    var buffer = Buffer.concat( this._buffers )
-    var offset = 0
+    if( this.currentRange == null ) {
+      debug( 'no current range' )
+      return next()
+    }
 
-    while( buffer.length - offset >= this.blockSize ) {
-      this._pushBlock( buffer.slice( offset, offset + this.blockSize ) )
-      offset = offset + this.blockSize
+    if( !this._rangeInChunk( chunk ) ) {
+      debug( 'chunk:ignore' )
```

Comment:
Yeah, that's the point – you pipe a readable stream into the BlockMap.FilterStream and it it'll filter out unmapped blocks.

One of the main use cases for this is when images from sources that cannot be randomly accessed – i.e. an image being decompressed or extracted from an archive, or an HTTP stream – which is the case in Etcher, as the only way to flash blockmapped images at the moment is to have them in a .zip with the metadata folder alongside it.